pkg/controller/gcp/compute/network.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package compute
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/crossplaneio/crossplane/gcp/apis/compute/v1alpha1"
+
+	"github.com/pkg/errors"
+	googlecompute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	computev1alpha1 "github.com/crossplaneio/crossplane/gcp/apis/compute/v1alpha1"
+	gcpapis "github.com/crossplaneio/crossplane/gcp/apis/v1alpha1"
+	gcpclients "github.com/crossplaneio/crossplane/pkg/clients/gcp"
+	"github.com/crossplaneio/crossplane/pkg/meta"
+	"github.com/crossplaneio/crossplane/pkg/resource"
+)
+
+const (
+	// Error strings.
+	errNewClient = "cannot create new Compute Service"
+	errNotVPC    = "managed resource is not a Network resource"
+
+	namePrefix = "vpc"
+)
+
+// NetworkController is the controller for Network CRD.
+type NetworkController struct{}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *NetworkController) SetupWithManager(mgr ctrl.Manager) error {
+	r := resource.NewManagedReconciler(mgr,
```

Comment:
The work you and @negz did on the shared managed reconciler makes adding new resource types so much cleaner :)